### PR TITLE
Don't generate #line directives for SimpleSite

### PR DESCRIPTION
**Pull request by mandolaerik**
- Revert previous SimpleSite change
- Polish site strings
- Don't generate #line directives for SimpleSite

### CODE REVIEW

**Comment by syssimics** [comment]
Verification [#10373509](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=10373509): fail

File: py/dml/toplevel.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def determine_version(filestr, filename):
         column = ver_start - filestr.rfind('\n', 0, ver_start)
         m = check_version.match(filestr, pos=ver_start)
         if not m:
-            raise ESYNTAX(SimpleSite(filename, lineno, column),
+            raise ESYNTAX(SimpleSite("%s:%d:%d" % (filename, lineno, column)),
```
Comment (lwaern-intel) [inline py/dml/toplevel.py:73]: You're being inconsistent about the use of f-strings vs `%` when creating `SimpleSite`s, I'm ok with either alternative, but alternating between both in the same PR for the same use-case is very strange to me.

File: py/dml/logging.py [hunk 2]
```diff
@@ -276,17 +276,14 @@ def bitorder(self): pass
...
+    def __init__(self, name, dml_version=(1, 4)):
         self._name = name
         self._dml_version = dml_version
-        self._line = '' if line is None else f':{line}'
-        if column is not None:
-            self._line += f':{column}'
     def __repr__(self):
-        return f'<site {self.loc()}>'
+        return '<site %s>' % (self._name,)
     def loc(self):
```
Comment (lwaern-intel) [inline py/dml/logging.py:285]: Why move from f-string to `%`?

**Review by lwaern-intel — CHANGES_REQUESTED** [review]
Address the `%` vs f-string dilemma and pick one.

File: py/dml/toplevel.py (hunk 1 above)
Comment (mandolaerik) [reply py/dml/toplevel.py:73]: This comes from a commit that is a pure revert, I didn't bother to preserve the f-string migration parts of the reverted commit.

Comment (mandolaerik) [reply py/dml/toplevel.py:73]: (in general, f-string should be the default choice that should be used unless there's a good reason why % is better.)

**Comment by syssimics** [comment]
Verification [#10378786](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=10378786): pass

**Comment by syssimics** [comment]
Verification [#10379061](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=10379061): pass

**Review by lwaern-intel — APPROVED** [review]

**Pull request merged**